File: data-structures/stacks/stack.cpp
#include <vector>
#include <iostream>

using namespace std;

class Stack {
  public: int top = 0;

  public: vector<int> arr = {3};

  int push(int x) {
    top = top + 1;
    return arr[top] = x;
  }

  int pop() {
    if (top - 1 >= 0) {
      top = top - 1;
      return arr[top + 1];
    } else {
      return arr[top];
    }
  }

  vector<int> data() {
    return arr;
  }
};

int main() {

  Stack st;

  st.push(2);
  st.push(1);

  cout << st.pop();
  cout << st.pop();
  cout << st.pop();

  return 0;
}

File: data-structures/stacks/stack.js
class Stack {
  static arr = [];
  static top = 0;

  pop() {
    if (Stack.top > 0) {
      Stack.top = Stack.top - 1;
      return Stack.arr[Stack.top + 1];
    } else {
      return Stack.arr[Stack.top];
    }
  }

  push(x) {
      if (Stack.arr.length > 0) {
        Stack.top = Stack.top + 1
        Stack.arr[Stack.top] = x;
      } else {
        Stack.arr[Stack.top] = x;
      }
  }
}

const stack = new Stack();

stack.push(1);
stack.push(2);
stack.push(3);
stack.push(4);

console.log(Stack.arr);
console.log(stack.pop());
console.log(stack.pop());
console.log(stack.pop());
console.log(stack.pop());
